internal/bminventory/inventory_test.go

```diff
@@ -8887,100 +8902,71 @@ var _ = Describe("UpdateDiscoveryIgnition", func() {
 		ctrl.Finish()
 	})
 
-	It("saves the given string to the cluster", func() {
+	It("saves the given string to InfraEnv", func() {
 		override := `{"ignition": {"version": "3.1.0"}, "storage": {"files": [{"path": "/tmp/example", "contents": {"source": "data:text/plain;base64,aGVscGltdHJhcHBlZGluYXN3YWdnZXJzcGVj"}}]}}`
-		params := installer.UpdateDiscoveryIgnitionParams{
-			ClusterID:               clusterID,
-			DiscoveryIgnitionParams: &models.DiscoveryIgnitionParams{Config: override},
-		}
-		mockS3Client.EXPECT().DeleteObject(gomock.Any(),
-			fmt.Sprintf("%s.iso", fmt.Sprintf(s3wrapper.DiscoveryImageTemplate, clusterID.String()))).Return(false, nil)
-		mockEvents.EXPECT().SendClusterEvent(gomock.Any(), eventstest.NewEventMatcher(
-			eventstest.WithNameMatcher(eventgen.DiscoveryIgnitionConfigAppliedEventName),
-			eventstest.WithClusterIdMatcher(clusterID.String())))
-		response := bm.UpdateDiscoveryIgnition(ctx, params)
-		Expect(response).To(BeAssignableToTypeOf(&installer.UpdateDiscoveryIgnitionCreated{}))
+		params := installer.UpdateInfraEnvParams{
+			InfraEnvID:           *infraEnv.ID,
+			InfraEnvUpdateParams: &models.InfraEnvUpdateParams{IgnitionConfigOverride: override},
+		}
+		mockIgnitionBuilder.EXPECT().FormatDiscoveryIgnitionFile(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(discovery_ignition_3_1, nil).Times(2)
+		mockS3Client.EXPECT().GetBaseIsoObject(gomock.Any(), gomock.Any()).Return("rhcos", nil).Times(1)
+		mockS3Client.EXPECT().UploadISO(gomock.Any(), gomock.Any(), "rhcos", gomock.Any()).Return(nil).Times(1)
```

Comment:
Not for this PR, but this isn't how this works in any environment any more.
After the v1 cleanup we should change the unit tests to use the image service everywhere.

Reply:
Agreed. I just opened [MGMT-9460](https://issues.redhat.com/browse/MGMT-9460) to capture this.